### PR TITLE
Added Manga To Crunchyroll & Added Watch Episode Buttons

### PR DESCRIPTION
**Pull request by ScarVite**
POW:

Discord_hSm2N0779a
Discord_AudmUOOJSI
Discord_lbdBV0nMrm

### CODE REVIEW

File: websites/C/Crunchyroll/presence.ts [hunk 1]
```diff
@@ -32,7 +36,16 @@ if (lastPlaybackState != playback) {
 
 var iFrameVideo: any, currentTime: any, duration: any, paused: any;
```
Comment (ririxidev) [inline websites/C/Crunchyroll/presence.ts:37]: Well no `var` nor `any` type

File: websites/C/Crunchyroll/presence.ts [hunk 2]
```diff
@@ -48,7 +61,52 @@ presence.on("UpdateData", async () => {
...
+      presenceData.details = (await strings).browse;
+      presenceData.startTimestamp = browsingStamp;
+
+      delete presenceData.state;
+      delete presenceData.smallImageKey;
+
+      reading = true;
+    }
+
+    presence.setActivity(presenceData, reading);
```
Comment (ririxidev) [inline websites/C/Crunchyroll/presence.ts:106]: Don't need to set second argument so please remove `let reading` variable too

**Review by ririxidev — CHANGES_REQUESTED** [review]
remove `getTimestamps` function from code with comments and use `presence.getTimestamps()` from `Presence` class
and if you are editing code already remove on line `135`, `presenceData.startTimestamp = timestamps[0]` thank you.

**Review by ririxidev — APPROVED** [review]
thank you 😃  lgtm

File: websites/C/Crunchyroll/presence.ts [hunk 3]
```diff
@@ -48,7 +47,48 @@ presence.on("UpdateData", async () => {
...
 
-  if (!playback) {
+  if (!playback && document.location.pathname.includes("/manga")) {
+
+    if(document.location.pathname.includes("/read")) {
+      const title = document.querySelector(".chapter-header a").innerHTML,
+      currChapter = document.querySelector(".chapter-header").innerHTML.split("</a>")[1].split("\n")[0];
+      let currPage = document.querySelector(".first-page-number").innerHTML;
+      currPage = currPage == "" ? "1" : currPage;
+      const lastPage = document.querySelector(".images").children.length;
```
Comment (ririxidev) [inline websites/C/Crunchyroll/presence.ts:57]:
```suggestion
      const title = document.querySelector(".chapter-header a").innerHTML,
      currChapter = document.querySelector(".chapter-header").innerHTML.split("</a>")[1].split("\n")[0],
      lastPage = document.querySelector(".images").children.length,
      currPage = document.querySelector(".first-page-number").innerHTML === "" ? "1" : currPage;
```

File: websites/C/Crunchyroll/presence.ts [hunk 4]
```diff
@@ -59,34 +99,31 @@ presence.on("UpdateData", async () => {
...
-    episod = document.querySelectorAll("#showmedia_about_media h4");
-    epName = document.querySelector("h4#showmedia_about_name");
-    episode = episod[1].innerText + " - " + epName.innerText;
+    const videoTitle = document.querySelector(".ellipsis .text-link span"),
+    episod = document.querySelectorAll("#showmedia_about_media h4"),
+    epName = document.querySelector("h4#showmedia_about_name"),
+    episode = episod[1].innerHTML + " - " + epName.innerHTML;
 
-    var timestamps = getTimestamps(
+    const timestamps = presence.getTimestamps(
```
Comment (ririxidev) [inline websites/C/Crunchyroll/presence.ts]: merge with const above dx

**Review by ririxidev — CHANGES_REQUESTED** [review]
well on line 1 `var` -> `const`

**Comment by ScarVite** [comment]
But the thing i don't get, is codacy only looking at changed lines?

because it never showed any of these issues before, especially with that "any" type stuff

**Review by ririxidev — COMMENTED** [review]
Well, codacy is weird that's why we are moving to deepscan soon

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**